src/main/java/com/code_intelligence/jazzer/junit/SeedSerializer.java

```diff
@@ -0,0 +1,142 @@
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+interface SeedSerializer {
+  Object[] read(byte[] bytes);
+  default boolean allReadsValid() {
+    return true;
+  }
+
+  byte[] write(Object[] args) throws UnsupportedOperationException;
+
+  /**
+   * Creates specialized {@link SeedSerializer} instances for the following method parameters:
+   * <ul>
+   *   <li>{@code byte[]}
+   *   <li>{@code FuzzDataProvider}
+   *   <li>Any other types will attempt to be created using either Autofuzz or the experimental
+   * mutator framework if {@link Opt}'s {@code experimentalMutator} is set.
+   * </ul>
+   */
+  static SeedSerializer of(Method method) {
+    if (method.getParameterCount() == 0) {
+      throw new IllegalArgumentException(
+          "Methods annotated with @FuzzTest must take at least one parameter");
+    }
+    if (method.getParameterCount() == 1 && method.getParameterTypes()[0] == byte[].class) {
+      return new ByteArraySeedSerializer();
+    } else if (method.getParameterCount() == 1
+        && method.getParameterTypes()[0] == FuzzedDataProvider.class) {
+      return new FuzzedDataProviderSeedSerializer();
+    } else {
+      Optional<ArgumentsMutator> argumentsMutator =
+          Opt.experimentalMutator ? ArgumentsMutator.forMethod(method) : Optional.empty();
+      return argumentsMutator.<SeedSerializer>map(ArgumentsMutatorSeedSerializer::new)
+          .orElseGet(() -> new AutofuzzSeedSerializer(method));
+    }
+  }
+}
+
+final class ByteArraySeedSerializer implements SeedSerializer {
+  @Override
+  public Object[] read(byte[] bytes) {
+    return new Object[] {bytes};
+  }
+
+  @Override
+  public byte[] write(Object[] args) {
+    return (byte[]) args[0];
+  }
+}
+
+final class FuzzedDataProviderSeedSerializer implements SeedSerializer {
+  @Override
+  public Object[] read(byte[] bytes) {
+    return new Object[] {FuzzedDataProviderImpl.withJavaData(bytes)};
+  }
+
+  @Override
+  public byte[] write(Object[] args) throws UnsupportedOperationException {
+    // While we could get the underlying bytes, it's not possible to provide Java seeds for fuzz
+    // tests with a FuzzedDataProvider parameter.
+    throw new UnsupportedOperationException();
+  }
+}
+
+final class ArgumentsMutatorSeedSerializer implements SeedSerializer {
+  private final ArgumentsMutator mutator;
+  private boolean allReadsValid;
+
+  public ArgumentsMutatorSeedSerializer(ArgumentsMutator mutator) {
+    this.mutator = mutator;
+  }
+
+  @Override
+  public Object[] read(byte[] bytes) {
+    allReadsValid &= mutator.read(new ByteArrayInputStream(bytes));
+    return mutator.getArguments();
+  }
+
+  @Override
+  public boolean allReadsValid() {
+    return allReadsValid;
+  }
+
+  @Override
+  public byte[] write(Object[] args) {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    mutator.writeAny(out, args);
```

Comment:
Is this safe? Could user provided argument arrays vary in size?

Reply:
JUnit checks the parameter count and types, which will catch such issues before they reach this method (I added comments).